pymeasure/experiment/results.py

```diff
@@ -269,7 +269,7 @@ def units_found(parameter, units):
             if parameter.name in parameters:
                 value, units = parameters[parameter.name]
 
-                if units_found(parameter, units):
+                if units_found(parameter, units) or (type(parameter) == Parameter and units is not None):
```

Comment:
`procedure.parameter_objects()` returns a dict of Parameters, so is it actually possible that `parameter` is not a `Parameter`? Also, typically it's preferable to use `isinstance`  to check object type - or are you intending to check for the base class only?

Reply:
Yes, I intend to check for a base class only (hence it is indeed possible to have a `parameter` that is not a `Parameter` but rather one of the subclasses) since I think only in the `Parameter` base class it makes sense to have a space somewhere in the value.

Reply:
>  parameter that is not a Parameter but rather one of the subclasses

Hold on, though, a `BooleanParameter`, being a subclass, *is a* `Parameter` (see behaviour of `isinstance()`)

A behaviour that is valid for a base class, but not a subclass, looks like a code smell to me. I realise this might be hard to change now, but we should try to not compound/rely on that too much if possible.

Reply:
I completely agree on that, but I couldn't think of a better approach.
An alternative would by to replace `type(parameter) == Parameter` by `not hasattr(parameter, 'units')` such that value and units are combined only when the `Parameter` has not units; in that case the full string is also passed to `BooleanParameter`, though you might expect that this never has any consequence as a string for the `BooleanParameter` should in principle always be `False` or `True`.

Another way would be to always pass the full string (so probably not even separate the value and unit from the string) and only perform the separation within the Parameter subclasses where it makes sense.

Reply:
puh, no idea to be honest :-P